carapace-server/src/main/java/org/carapaceproxy/server/certificates/ocsp/OcspStaplingManager.java

```diff
@@ -0,0 +1,194 @@
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.bouncycastle.asn1.ocsp.OCSPResponseStatus;
+import org.bouncycastle.cert.ocsp.BasicOCSPResp;
+import org.bouncycastle.cert.ocsp.CertificateStatus;
+import org.bouncycastle.cert.ocsp.OCSPException;
+import org.bouncycastle.cert.ocsp.OCSPReq;
+import org.bouncycastle.cert.ocsp.OCSPResp;
+import org.bouncycastle.cert.ocsp.SingleResp;
+import org.carapaceproxy.cluster.GroupMembershipHandler;
+import org.carapaceproxy.configstore.ConfigurationStore;
+import org.glassfish.jersey.internal.guava.ThreadFactoryBuilder;
+
+/**
+ *
+ * @author paolo.venturi
+ */
+public class OcspStaplingManager implements Runnable {
+
+    public static final String THREAD_NAME = "ocsp-stapling-manager";
+
+    private static final Logger LOG = Logger.getLogger(OcspStaplingManager.class.getName());
+    private ScheduledExecutorService scheduler;
+    private ScheduledFuture<?> scheduledFuture;
+    private volatile boolean started; // keep track of start() calling
+    private ConfigurationStore store;
+    private volatile int period = 0; // in seconds
+    private GroupMembershipHandler groupMembershipHandler;
+
+    private final ConcurrentHashMap<String, OcspCheck> ocspChecks = new ConcurrentHashMap<>();
+
+    private static final class OcspCheck {
+
+        Certificate[] chain;
+        OCSPResp response;
+
+        public OcspCheck(Certificate[] chain, OCSPResp response) {
+            this.chain = chain;
+            this.response = response;
+        }
+    }
+
+    public synchronized void start() {
+        started = true;
+        if (period <= 0) {
+            return;
+        }
+        if (scheduler == null) {
+            scheduler = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder().setNameFormat(THREAD_NAME).build());
+        }
+
+        //LOG.info("Starting " + OcspStaplingManager.class.getName() + ", period: " + period + " seconds" + (TESTING_MODE ? " (TESTING_MODE)" : ""));
+        scheduledFuture = scheduler.scheduleWithFixedDelay(this, 0, period, TimeUnit.SECONDS);
+    }
+
+    public synchronized void stop() {
+        started = false;
+        if (scheduler != null) {
+            scheduler.shutdown();
+            try {
+                scheduler.awaitTermination(10, TimeUnit.SECONDS);
+                scheduler = null;
+                scheduledFuture = null;
+            } catch (InterruptedException err) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    @Override
+    public void run() {
+        if (groupMembershipHandler != null) {
+            int acquirePeriod = period;
+            if (acquirePeriod <= 0) {
+                acquirePeriod = 1;
+            }
+            groupMembershipHandler.executeInMutex(THREAD_NAME, acquirePeriod, () -> {
```

Comment:
This is not needed.
In current design (AFAIK) ocsp response is local, it is not shared